src/query/functions/temporal/base.go

```diff
@@ -223,14 +229,31 @@ func batchProcess(
 	return multiErr.FinalError()
 }
 
-func buildBlockBatch(
+func parallelProcess(
+	ctx context.Context,
 	idx int,
 	iter block.SeriesIter,
 	builder block.Builder,
 	blockMeta blockMeta,
 	processor processor,
 	mu *sync.Mutex,
 ) error {
+	var (
+		start      = time.Now()
+		decodeTime time.Duration
+	)
+	_, sp, _ := xcontext.StartSampledTraceSpan(ctx, tracepoint.TemporalDecodeParallel)
+	defer func() {
+		if decodeTime == 0 {
+			return // Do not record this span if instrumentation is not turned on.
```

Comment:
Is there anything housekeeping we need to do with the unfinished unused span here?

Reply:
Changed so we only create the span if we intend to finish it